### PR TITLE
This creates configurable number of disks with configurable size for centos72

### PR DESCRIPTION
**Pull request by dwaiba**
### Contributing guide
https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md

### Changelog
* This creates configurable number of disks with configurable size for centos72
*
*
*
*

### Description of the change

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@dwaiba__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (dwaiba). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by dwaiba** [comment]
This would really assist many trying to automate for centos.. It exists for Windows but not for Linux end to end automated

**Comment by singhkay** [comment]
@dwaiba This is really useful. How about we make this a generic Linux template that can deploy into any Linux flavor based on a parameter and allowed set of values? Currently the only value would be CentOS and would give other folks a platform to add more Linux OSes over time.

**Comment by dwaiba** [comment]
Added the parameters file
@singhkay Only thing required is adding Ubuntu Offers and publisher (Canonical) with azuredeploy.sh with apt-get. All tools remain same. This is an offshoot of a dynadisk/slurm/torque/IBM lsf master/slaves with Intel Lustre end to end template that I am working on for CentOS-HPC (working on the cross compiler script as kernel for the same is custom with Intel MPI Pre-installed). I will update this single node template later if it is ok. Since many require this out of the box, it would be great if it is now merged with the param file already commited to the forked master

**Comment by singhkay** [comment]
@dwaiba Sure, ok with mergin this now but let's name this 201-linux-dynamic-disks so that the URL doesn't change when the updates for other OSes are made later. 

Also see the error ` AssertionError: centos_dynadisk/metadata.json - #/itemDisplayName:Failed "maxLength" criteria`

**Comment by dwaiba** [comment]
Renamed 
Failing with the following as merged endstate URL is not valid during build

       "error": "InvalidContentLink : Unable to download deployment content from 'https://raw.githubusercontent.com/azure/azure-quickstart-templates/master/201-vm-linux-dynamic-data-disks/diskSelection.json'. The tracking Id is 'dd507aff-36e3-40e4-b6ac-60d5c175eb33'. Please see http://aka.ms/arm-deploy for usage details.\n"

**Comment by dwaiba** [comment]
Requesting merge 👍

File: 201-vm-linux-dynamic-data-disks/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# centos_dynadisk
```
Comment (singhkay) [inline 201-vm-linux-dynamic-data-disks/README.md]: Rename to "Setup Linux Dynamic data disks" or something that explains a bit about the scenario

File: 201-vm-linux-dynamic-data-disks/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fazure-quickstart-templates%2Fmaster%2F201-vm-linux-dynamic-data-disks%2Fazuredeploy.json" target="_blank">
+   <img alt="Deploy to Azure" src="http://azuredeploy.net/deploybutton.png"/>
+</a>
```
Comment (singhkay) [inline 201-vm-linux-dynamic-data-disks/README.md]: Needs a visualize button too

File: 201-vm-linux-dynamic-data-disks/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,354 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "dnsPrefix": {
```
Comment (singhkay) [inline 201-vm-linux-dynamic-data-disks/azuredeploy.json]: Should be dnsLabelPrefix for consistency. See #966

File: 201-vm-linux-dynamic-data-disks/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+      "type": "Microsoft.Compute/virtualMachines/extensions",
+      "name": "[concat(parameters('masterVMName'), '/Installation')]",
+      "location": "[resourceGroup().location]",
+      "dependsOn": [
+        "[concat('Microsoft.Compute/virtualMachines/', parameters('masterVMName'))]"
+      ],
+      "properties": {
+        "publisher": "Microsoft.OSTCExtensions",
+        "type": "CustomScriptForLinux",
+        "typeHandlerVersion": "1.3",
```
Comment (singhkay) [inline 201-vm-linux-dynamic-data-disks/azuredeploy.json]: I think 1.5 is the latest. Also use autoupgrademinorversion property

**Comment by singhkay** [comment]
@dwaiba URLs are case sensitive. Use "Azure" instead of "azure" for the CI to replace the URL with your fork. See https://github.com/Azure/azure-quickstart-templates/tree/master/1-CONTRIBUTION-GUIDE#travis-ci

**Comment by dwaiba** [comment]
All Changes in place..
Replacement to fork is not kicking in for Azure
<code>
 "error": "InvalidContentLink : Unable to download deployment content from 'https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/201-vm-linux-dynamic-data-disks/diskSelection.json'.
</code>

**Comment by dwaiba** [comment]
@singhkay Build success and tested multiple times end to end. Seems CI does not replace "Azure" with "githubUsername" during build. Requesting merge. All other changes are in.

**Comment by dwaiba** [comment]
@singhkay Would it be possible to have a look #1900 . It is working fine ...

File: 201-vm-linux-dynamic-data-disks/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+      "name": "[concat(parameters('masterVMName'), '/Installation')]",
+      "location": "[resourceGroup().location]",
+      "dependsOn": [
+        "[concat('Microsoft.Compute/virtualMachines/', parameters('masterVMName'))]"
+      ],
+      "properties": {
+        "publisher": "Microsoft.OSTCExtensions",
+        "type": "CustomScriptForLinux",
+        "typeHandlerVersion": "1.5",
+        "AutoUpgradeMinorVersion": "true",
```
Comment (singhkay) [inline 201-vm-linux-dynamic-data-disks/azuredeploy.json]: this is a boolean like here https://github.com/Azure/azure-quickstart-templates/blob/c1ebf2180c6c68f5ac5630b6ace253abb320789c/sqlvm-alwayson-cluster/nested/configuringBackupADVM.json#L37

**Comment by singhkay** [comment]
@dwaiba looks good, one small change needed

**Comment by dwaiba** [comment]
@singhkay #1900 build passed on last commit  for 
https://github.com/Azure/azure-quickstart-templates/pull/1900/commits/ceb65794c582cdd98dce04ff933455ddbac250fc `"autoUpgradeMinorVersion":true,`

**Comment by singhkay** [comment]
lgtm, merged

**Pull request merged**